app/controllers/campaigns_controller.rb

```diff
@@ -106,6 +108,10 @@ def programs
     set_page
     set_presenter
     @search_terms = params[:courses_query]
+    if @search_terms.present?
+      @results = @presenter.search_courses(@search_terms)
+      @num_of_results = @results.count
```

Comment:
I don't think having a separate instance variable for the count is necessary. You can just use `results.count` or similar in the views.